test/spix/smoke_test.py

```diff
@@ -22,31 +24,115 @@ def screenshot_path():
     return img_path
 
 
+@pytest.fixture
+def diff_path():
+    """
+    Returns the path for a folder to put image comparison errors into.
+    Supposed to be in the path where pytest-html writes its report.
+    """
+    img_path = os.path.join(os.getcwd(), "report", "diffs")
+    print(f"Images will be written to {img_path}")
+    os.makedirs(img_path, exist_ok=True)
+    return img_path
+
+
+@pytest.fixture
+def screenshot_check(image_diff, image_diff_dir, screenshot_path, diff_path, extra):
+    def inner(test_name, image_name):
+        result = image_diff(
+            os.path.join(screenshot_path, "{}.png".format(image_name)),
+            str(
+                Path(__file__).parent.parent
+                / "testdata"
+                / "control_images"
+                / test_name
+                / "expected_{}.png".format(image_name)
+            ),
+            threshold=0.05,
+            suffix=image_name,
+            throw_exception=False,
+        )
+        if os.path.exists(result.diff_path):
+            shutil.copyfile(
+                result.diff_path,
+                os.path.join(diff_path, "{}_diff.png".format(image_name)),
+            )
+            extra.append(
+                extras.html(
+                    "Screenshot check for {}: diff value {}, threshold {}".format(
+                        image_name, result.value, result.thresh
+                    )
+                )
+            )
+            extra.append(
+                extras.html('<img src="diffs/{}_diff.png"/>'.format(image_name))
+            )
+        return result.value < result.thresh
+
+    return inner
+
+
 def test_start_app(app, screenshot_path, extra, process_alive):
     """
     Starts a test app to the welcome screen and creates a screenshot.
     """
     assert app.existsAndVisible("mainWindow")
 
-    time.sleep(2)
+    time.sleep(1)
 
     app.takeScreenshot("mainWindow", os.path.join(screenshot_path, "startup.png"))
     assert process_alive()
     extra.append(extras.html('<img src="images/startup.png"/>'))
 
 
 @pytest.mark.project_file("test_wms.qgz")
-def test_message_logs(app, screenshot_path, extra, process_alive):
+def test_wms_layer(app, screenshot_path, screenshot_check, extra, process_alive):
+    """
+    Starts a test app and check for WMS layer support (including rendering check and message logs).
+    This also tests that QField is able to reach QGIS's crucial srs.db.
+    """
+    assert app.existsAndVisible("mainWindow")
+
+    time.sleep(4)
```

Comment:
Wait for what?

Reply:
Waiting for the world to be a better place ;) it's an arbitrary wait ATM as we have no means to be informed of the status of project load not can we plug into a signal.

It's something that'd be nice for spix to integrate, some form of signal spy/watcher. That said, with the new spix dev, we can trigger QML functions. In a separate PR I'm thinking of doing a while loop and can a function in qgismobileapp.cpp to check if a project has been loaded (and had a successful first draw).

For the time being, 4 sec wait works :)

Reply:
Can you add a comment?